internal/providers/amazon/eks/driver/cluster_creator.go

```diff
@@ -0,0 +1,511 @@
+			VpcCidr:            aws.StringValue(modelCluster.VpcCidr),
+			ScaleEnabled:       commonCluster.GetScaleOptions() != nil && commonCluster.GetScaleOptions().Enabled,
+			DefaultUser:        modelCluster.DefaultUser,
+			ClusterRoleID:      modelCluster.ClusterRoleId,
+			NodeInstanceRoleID: modelCluster.NodeInstanceRoleId,
+			KubernetesVersion:  modelCluster.Version,
+			LogTypes:           modelCluster.LogTypes,
+		},
+		PostHooks:        createRequest.PostHooks,
+		OrganizationName: org.Name,
+	}
+
+	for _, mode := range modelCluster.APIServerAccessPoints {
+		switch mode {
+		case "public":
+			input.EndpointPublicAccess = true
+		case "private":
+			input.EndpointPrivateAccess = true
+		}
+	}
+
+	subnets := make([]workflow.Subnet, 0)
+	subnetMapping := make(map[string][]workflow.Subnet)
+	for _, eksSubnetModel := range modelCluster.Subnets {
+		subnet := workflow.Subnet{
+			SubnetID:         aws.StringValue(eksSubnetModel.SubnetId),
+			Cidr:             aws.StringValue(eksSubnetModel.Cidr),
+			AvailabilityZone: aws.StringValue(eksSubnetModel.AvailabilityZone)}
+
+		subnets = append(subnets, subnet)
+
+		nodePools := getNodePoolsForSubnet(commonCluster.(*cluster.EKSCluster).GetSubnetMapping(), subnet)
+		c.logger.Debugf("node pools mapped to subnet %s: %v", subnet.SubnetID, nodePools)
+
+		for _, np := range nodePools {
+			subnetMapping[np] = append(subnetMapping[np], subnet)
+		}
+	}
+
+	input.Subnets = subnets
+	input.ASGSubnetMapping = subnetMapping
+
+	asgList := make([]workflow.AutoscaleGroup, 0)
+	for _, np := range modelCluster.NodePools {
+		asg := workflow.AutoscaleGroup{
+			Name:             np.Name,
+			NodeSpotPrice:    np.NodeSpotPrice,
+			Autoscaling:      np.Autoscaling,
+			NodeMinCount:     np.NodeMinCount,
+			NodeMaxCount:     np.NodeMaxCount,
+			Count:            np.Count,
+			NodeImage:        np.NodeImage,
+			NodeInstanceType: np.NodeInstanceType,
+			Labels:           np.Labels,
+		}
+		asgList = append(asgList, asg)
+	}
+
+	input.AsgList = asgList
+
+	labelsMap, err := cluster.GetDesiredLabelsForCluster(ctx, commonCluster, nil, false)
+	if err != nil {
+		_ = commonCluster.SetStatus(pkgCluster.Error, "failed to get desired labels")
+
+		return nil, err
+	}
+	input.NodePoolLabels = labelsMap
+
+	workflowOptions := client.StartWorkflowOptions{
+		TaskList:                     "pipeline",
+		ExecutionStartToCloseTimeout: 1 * 24 * time.Hour,
+	}
+	exec, err := c.workflowClient.ExecuteWorkflow(ctx, workflowOptions, cluster.EKSCreateClusterWorkflowName, input)
+	if err != nil {
+		return nil, err
+	}
+
+	err = commonCluster.(*cluster.EKSCluster).SetCurrentWorkflowID(exec.GetID())
+	if err != nil {
+		return nil, err
+	}
+
+	timer, err := getClusterStatusChangeMetricTimer(commonCluster.GetCloud(), commonCluster.GetLocation(), pkgCluster.Creating, commonCluster.GetOrganizationId(), commonCluster.GetName(), c.statusChangeDurationMetric)
+	if err != nil {
+		return nil, err
+	}
+	go func() {
```

Comment:
This won't track the duration of an EKS cluster creation if Pipeline is restarted.